packages/api/src/command/medical/patient/settings/create-patient-settings.ts

```diff
@@ -138,50 +161,52 @@ export async function upsertPatientSettingsForCx({
 }: PatientSettingsUpsertForCxProps): Promise<PatientSettingsUpsertResults> {
   const { log } = out(`upsertPatientSettingsForCx - cx ${cxId}`);
 
-  const totalPatients = await getPatientsCount({ cxId, facilityId });
-  if (totalPatients === 0) {
+  const patientIds = await getPatientIds({ cxId, facilityId });
+  if (patientIds.length === 0) {
     log(`No patients found for cx ${cxId}`);
     return { patientsFoundAndUpdated: 0 };
   }
 
   let processedTotal = 0;
   let failedTotal = 0;
-  let fromItem: string | undefined;
-
-  do {
-    const { currPageItems, nextPageItemId } = await getPaginationItems(
-      { count: BATCH_SIZE, ...(fromItem ? { fromItem } : {}) },
-      pagination => getPatients({ cxId, facilityId, pagination }),
-      () => Promise.resolve(totalPatients)
-    );
-
-    if (currPageItems.length === 0) break;
+  const failedIds: string[] = [];
 
-    const patientIds = currPageItems.map(p => p.id);
-    const { processedCount, failedCount, error } = await processBatch({
-      patients: patientIds,
+  for (let i = 0; i < patientIds.length; i += BATCH_SIZE) {
+    const batchIds = patientIds.slice(i, i + BATCH_SIZE);
```

Comment:
No need to update this now, but it's easier (and cleaner) to use Lodash's [.chunk()](https://lodash.com/docs/4.17.15#chunk) - recommending this because we use Lodash throughout the codebase, including a bunch of usages of `.chunk()`